### PR TITLE
[S3] Allow configuring AWS SDK's logging level

### PR DESCRIPTION
**Pull request by rahul003**
- Allows configuring the AWS CPP SDK's logging level correctly through the env variable `AWS_LOG_LEVEL`. This helps suppress verbose logs and allows access to debugging logs in the case of issues.
- The current behavior takes in the TF log level and passes that along to the SDK. That has a few limitations. The logging levels don't map directly between AWS SDK (trace/debug/info/warning/error/fatal/off) and TF (info/warn/error). By passing TF level to SDK, we can never enable Debug and Trace. 
- Setting the AWS SDK's log level to Fatal by default. This is because in the regular flow of TF operations, many safe errors are seen. These confuse users into thinking their program is failing. Setting this to FATAL suppresses them.
- In addition to the above, this PR adds VLOG statements helpful for debugging S3 issues.

### CODE REVIEW

**Comment by gbaned** [comment]
@rahul003 Can you please resolve conflicts? Thanks!

**Comment by rahul003** [comment]
@gbaned Fixed the conflict, please take a look

File: tensorflow/core/platform/s3/s3_file_system.cc [hunk 1]
```diff
@@ -177,6 +177,7 @@ class S3RandomAccessFile : public RandomAccessFile {
 
   Status Read(uint64 offset, size_t n, StringPiece* result,
               char* scratch) const override {
+    VLOG(1) << "ReadFilefromS3 s3://" << bucket_ << "/" << object_ << " from " << offset << " for n:" << n;
```
Comment (mihaimaruseac) [inline tensorflow/core/platform/s3/s3_file_system.cc]: Instead of `1` we should use one of the constants to make this readable

**Review by mihaimaruseac — CHANGES_REQUESTED** [review]

File: tensorflow/core/platform/s3/s3_file_system.cc (hunk 1 above)
Comment (rahul003) [reply tensorflow/core/platform/s3/s3_file_system.cc]: Will do, pending the discussion here: https://github.com/tensorflow/tensorflow/pull/36352#discussion_r382237079

Comment (mihaimaruseac) [reply tensorflow/core/platform/s3/s3_file_system.cc]: Let's keep them both as they are

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by rahul003** [comment]
Both the test failures are from unrelated changes. I know one has been fixed on master. Other probably hasn't. I'm not sure.

**Comment by mihaimaruseac** [comment]
Can you solve conflicts please?

**Comment by rahul003** [comment]
@mihaimaruseac Please check

File: tensorflow/core/platform/s3/s3_file_system.cc [hunk 2]
```diff
@@ -143,7 +144,7 @@ Aws::Client::ClientConfiguration& GetDefaultClientConfig() {
     if (ca_path) {
       cfg.caPath = Aws::String(ca_path);
     }
-
+
```
Comment (mihaimaruseac) [inline tensorflow/core/platform/s3/s3_file_system.cc]: Please don't insert trailing whitespace

**Review by mihaimaruseac — DISMISSED** [review]

**Comment by rahul003** [comment]
Removed the trailing whitespace

**Review by mihaimaruseac — APPROVED** [review]

**Comment by mihaimaruseac** [comment]
Can you take a look at the build failures please? Probably resync to master and run them again?

**Comment by rahul003** [comment]
Failures seemed unrelated to the PR. Pulled from master

**Review by mihaimaruseac — APPROVED** [review]

**Pull request merged**